cirq-core/cirq/testing/consistent_decomposition.py

```diff
@@ -47,3 +47,20 @@ def assert_decompose_is_consistent_with_unitary(val: Any, ignoring_global_phase:
     else:
         # coverage: ignore
         np.testing.assert_allclose(actual, expected, atol=1e-8)
+
+
+def assert_decompose_ends_at_default_gateset(val: Any):
+    """Ensures that all cirq gate decompositions end at the default cirq gateset."""
+
+    # pylint: disable=unused-variable
+    __tracebackhide__ = True
+    # pylint: enable=unused-variable
+    if protocols.is_parameterized(val):
+        return
```

Comment:
Doesn't decompose work for gates that have symbols in them too ? Maybe it would be best here to assert that this is being called on a parameter free `val` so one doesn't accidentally mistake accidentally putting in a symbol for success. WDYT ?

Reply:
We don't have general purpose analytical decompositions for parameterized gates which makes it much harder to ensure that the decompositions of parameterize gates ends up in the default gateset (eg: parameterized controlled gates and matrix gates). 


However, I gave some more thought to it and instead of ignoring all parameterized gates or all gates without a decompose implemented (done below), I've added a dedicated method `_known_gate_with_no_decomposition`, which should be used as an escape patch to ignore gates which are known to not have automatic decompositions into the target gateset. 

All cases (eg: parameterized controlled gates) for which we do not support automatic decomposition, will be added to this method in subsequent PRs.